### PR TITLE
[dv/otp_ctrl] otp reaches V2

### PR DESCRIPTION
**Pull request by cindychip**
This PR claims the V2 status for OTP_CTRL testbench.
This PR is depended on PR #6474 

Signed-off-by: Cindy Chen <chencindy@google.com>

### CODE REVIEW

File: hw/ip/otp_ctrl/doc/checklist.md [hunk 1]
```diff
@@ -162,26 +162,26 @@ Review        | [V2_CHECKLIST_SCOPED][]               | Done        |
...
+Tests         | [FPV_ALL_ASSERTIONS_WRITTEN][]          | N/A            |
+Tests         | [FPV_ALL_ASSUMPTIONS_REVIEWED][]        | N/A            |
+Tests         | [SIM_FW_SIMULATED][]                    | N/A            |
+Regression    | [SIM_NIGHTLY_REGRESSION_V2][]           | Done           |
+Coverage      | [SIM_CODE_COVERAGE_V2][]                | Done           |
+Coverage      | [SIM_FUNCTIONAL_COVERAGE_V2][]          | Done           |
+Coverage      | [FPV_CODE_COVERAGE_V2][]                | N/A            |
+Coverage      | [FPV_COI_COVERAGE_V2][]                 | N/A            |
+Issues        | [NO_HIGH_PRIORITY_ISSUES_PENDING][]     | Done           |
+Issues        | [ALL_LOW_PRIORITY_ISSUES_ROOT_CAUSED][] | Done           |
```
[the remark this answers is not in the document]
Comment (cindychip) [reply hw/ip/otp_ctrl/doc/checklist.md:182]: Thanks Sri, that is a good reminder, I will go over that with Michael on Thursday.

**Comment by cindychip** [comment]
Now this PR depends on PR #6515

**Comment by msfschaffner** [comment]
You may also want to double check coverage after the LC token hashing mechanism has been removed from OTP: #6558

**Review by matutem — COMMENTED** [review]
I don't know if it should go to V2 if the functional coverage plan is not ready (line 168). I thought the intent is to have the functional coverage review done to enter V2.

**Comment by cindychip** [comment]
Thank you for the commands. Yeah I should probably change that. This PR was
drafted before the functional coverage plan was merged. But a few issues
happened (mem_bkdr, coverage exclusion, etc) that this PR gets delayed. I
will try getting the functional coverage plan filled out and reviewed
before claiming V2 :)

  Cindy Chen

**Comment by cindychip** [comment]
Thank you Matute, please correct me if i was wrong. I do not think the functional coverage plan PR is merged? That is why I put it as an exception.

**Comment by cindychip** [comment]
now pending items:

- [ ] testplan second review
- [ ] Clean up OTP_CTRL with recent design change https://github.com/lowRISC/opentitan/pull/6558

**Review by matutem — APPROVED** [review]
LGTM

**Comment by cindychip** [comment]
Hey team @sriyerg @udinator @weicaiyang @matutem ,
From the nightly regression, I think OTP is ready to claim V2:
https://reports.opentitan.org/hw/ip/otp_ctrl/dv/latest/results.html
Could you take a look?

**Review by udinator — APPROVED** [review]
lgtm! congrats :)

**Review by weicaiyang — COMMENTED** [review]
LGTM, but I didn't join the review meeting, will let the others to approve this

**Review by sriyerg — APPROVED** [review]
Congratulations on achieving this milestone :)

**Comment by cindychip** [comment]
Thanks everyone for your support!!

**Comment by msfschaffner** [comment]
Congrats again for reaching this milestone!

**Pull request merged**